### PR TITLE
ENH make asynchronous email delivery

### LINKED ISSUE #526: Issue with mail server
The server can hang when sending an email to the participant.
With @agramfort, we think that it could come from the fact that to send an email, we connect to the mail server and send the email. However, this procedure is blocking the rest of the execution. We should migrate to asynchronous processing.

Some details with Flask integration are given there: https://stackoverflow.com/questions/40326651/flask-mail-sending-email-asynchronously-based-on-flask-cookiecutter

### PR DESCRIPTION
**Pull request by glemaitre**
closes #526 

Send email asynchronously.
I assume that tests will fail because we are now sending emails in different Python threads and they will not be registered in the main thread.

I am not sure how we can test it.

### CODE REVIEW

**Comment by glemaitre** [comment]
It might be possible to have to use `pytest-mock`

```python
mock_start = mocker.patch.object(threading.Thread, 'start', autospec=True)
...
mock_start.called
```

However, it will not fulfill some of the test cases.

File: ramp-frontend/ramp_frontend/utils.py [hunk 1]
```diff
@@ -55,10 +67,24 @@ def send_mail(to, subject, body):
     body : str
         The body of the email.
     """
+    app = current_app._get_current_object()
     try:
```
Comment (glemaitre) [inline ramp-frontend/ramp_frontend/utils.py:71]: I am unsure that having a `try/except` in to make async operation is a good idea.
@rth do you know if it is a blocking operation.

Comment (rth) [reply ramp-frontend/ramp_frontend/utils.py:71]: Yeah, it's not. So here you would just exit the clause and the error will never be handled. I think you would need to also add a try: except clause inside _send_async_email. to log a possible error. logging should be thread safe https://docs.python.org/3/howto/logging-cookbook.html#logging-from-multiple-threads

**Comment by rth** [comment]
Closing now that https://github.com/paris-saclay-cds/ramp-board/pull/557 was merged with similar functionality. But don't hesitate to re-open if needed.

**Pull request closed without merging**